### ISSUE TITLE
Possible type generation issue

### ISSUE THREAD

**Issue by Numpsy** [Zaid-Ajaj/Hawaii#10]
Hi,

An observation from looking at some generated code (I haven't had chance to look at it in any detail, it's just something I spotted).

I have a schema (generated by NSwag from an ASP.Net Core service) which contains this snippet in the components section:

```yaml
"ProblemDetails": {
  "type": "object",
  "additionalProperties": {
    "nullable": true
  },
  "properties": {
    "type": {
      "type": "string",
      "nullable": true
    },
    "title": {
      "type": "string",
      "nullable": true
    },
    "status": {
      "type": "integer",
      "format": "int32",
      "nullable": true
    },
    "detail": {
      "type": "string",
      "nullable": true
    },
    "instance": {
      "type": "string",
      "nullable": true
    },
    "extensions": {
      "type": "object",
      "nullable": true,
      "additionalProperties": {}
    }
  }
},
"ValidationProblemDetails": {
  "allOf": [
    {
      "$ref": "#/components/schemas/ProblemDetails"
    },
    {
      "type": "object",
      "additionalProperties": {
        "nullable": true
      },
      "properties": {
        "errors": {
          "type": "object",
          "nullable": true,
          "additionalProperties": {
            "type": "array",
            "items": {
              "type": "string"
            }
          }
        }
      }
    }
  ]
}
```

(```ProblemDetails``` and ```ValidationProblemDetails``` are classes built into ASP for error reporting fwiw, where the latter is a subclass of the former).

The current Hawaii code generator generates this:

```fsharp
type ProblemDetails = Map<string, Newtonsoft.Json.Linq.JToken>
type ValidationProblemDetails =
    { ``type``: Option<string>
      title: Option<string>
      status: Option<int>
      detail: Option<string>
      instance: Option<string>
      extensions: Option<Map<string, string>>
      errors: Option<Map<string, list<string>>> }
    ///Creates an instance of ValidationProblemDetails with all optional fields initialized to None. The required fields are parameters of this function
    static member Create (): ValidationProblemDetails =
        { ``type`` = None
          title = None
          status = None
          detail = None
          instance = None
          extensions = None
          errors = None }
```

and that bit with the ```Map<string, Newtonsoft.Json.Linq.JToken>``` looks wrong to me - it looks odd how ValidationProblemDetails contains all of the properties from ProblemDetails, but ProblemDetails is defined as a map?

Thanks.

**Comment by Zaid-Ajaj** [comment]
> that bit with the Map<string, Newtonsoft.Json.Linq.JToken> looks wrong to me

Yeah it is because of 
```json
  "additionalProperties": {
    "nullable": true
  }
```
which basically is saying: "this type is a dictionary where the values are anything" 

I agree that this is _probably_ not what you want. Can you please paste the full schema for me test with and try to make a nicer output out of this edge case? (I will ignore additionalProperties when it is not typed _and_ has typed properties)

**Comment by Numpsy** [comment]
The full schema is quite large and is from a product at work that I might not be able to publish in full, but i'll attach a minimal one that I just generated from an ASP/Swashbuckle test app - [swagger.txt](https://github.com/Zaid-Ajaj/Hawaii/files/6733960/swagger.txt) - that also generates Maps.

Running this one through Hawaii actually generates two map types rather than one map and one custom type, possibly because Swashbuckle and NSwag are using a different apporach to things when using ```allOf``` to represent inheritance?

**Comment by Zaid-Ajaj** [comment]
Fixed as of Hawaii v0.11.0 🚀 the schema you sent now generates these
```fs
namespace Swashbuckle.Types 

type ProblemDetails =
    { ``type``: Option<string>
      title: Option<string>
      status: Option<int>
      detail: Option<string>
      instance: Option<string> }
    ///Creates an instance of ProblemDetails with all optional fields initialized to None. The required fields are parameters of this function
    static member Create (): ProblemDetails =
        { ``type`` = None
          title = None
          status = None
          detail = None
          instance = None }

type ValidationProblemDetails =
    { ``type``: Option<string>
      title: Option<string>
      status: Option<int>
      detail: Option<string>
      instance: Option<string>
      errors: Option<Map<string, list<string>>> }
    ///Creates an instance of ValidationProblemDetails with all optional fields initialized to None. The required fields are parameters of this function
    static member Create (): ValidationProblemDetails =
        { ``type`` = None
          title = None
          status = None
          detail = None
          instance = None
          errors = None }

[<RequireQualifiedAccess>]
type GetTime =
    ///Success
    | OK of payload: int
    ///Bad Request
    | BadRequest of payload: ValidationProblemDetails
    ///Forbidden
    | Forbidden of payload: ProblemDetails
```
Can you give it a try on your bigger NSwag schema and see if the problem is resolved?

**Comment by Numpsy** [comment]
It looks ok in the new version :-)
